frontend/src/store.ts

```diff
@@ -0,0 +1,48 @@
+import { configureStore, createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+interface State {
+  isApiKeyModalOpen: boolean;
+  apiKey: string;
+  isMenuOpen: boolean;
+}
+
+const initialState: State = {
+  isApiKeyModalOpen: false,
+  apiKey: '',
+  isMenuOpen: false,

```

Comment:
IMO we should be only keeping that part of state in redux which is supposed to be shared across multiple components. I dont quite see how other components are going to need to know if menu is open or not. 

If i fail to see some usecase here, please LMK

Reply:
Both Navigation.tsx and APIKeyModal.tsx need access to isApiKeyModalOpen

apiKey might be able to be moved into the modal component if it is not going to be needed by Conversation.tsx when sending chats.

isMenuOpen is currently being used by both About.tsx and Conversation.tsx, and could be used by any other About-like pages in the future.